index.js

```diff
@@ -249,6 +251,100 @@ function getAPICredentials(cb) {
   });
 }
 
+/**
+ * Requests authorization to manage Apps Scrpit projects.
+ * @param {Boolean} useLocalhost True if a local HTTP server should be run
+ *     to handle the auth response. False if manual entry used.
+ */
+ function authorize(useLocalhost) {
+  var opts = {
+    access_type: 'offline',
+    scope: [
+      'https://www.googleapis.com/auth/script.deployments',
+      'https://www.googleapis.com/auth/script.projects',
+    ],
+  }
+  var authCode = useLocalhost ?
+    authorizeWithLocalhost(opts) :
+    authorizeWithoutLocalhost(opts);
+
+  authCode.then(code => {
+    return new Promise((res, rej) => {
+      oauth2Client.getToken(code, (err, token) => {
+        if (err) {
+          reject(err);
+        } else {
+          res(token);
+        }
+      })
+    })
+  })
+  .then(token => DOTFILE.RC.write(token))
+  .then(() => console.log(LOG.AUTH_SUCCESSFUL))
+  .catch(err => console.error(ERROR.ACCESS_TOKEN + err));
+}
+
+/**
+ * Requests authorization to manage Apps Scrpit projects. Spins up
+ * a temporary HTTP server to handle the auth redirect.
+ *
+ * @param {Object} opts OAuth2 options
+ * @return {Promise} Promise resolving with the authorization code
+ */
+function authorizeWithLocalhost(opts) {
+  return new Promise((resolve, reject) => {
+    var server = http.createServer(function(req, res) {
+      var url_parts = url.parse(req.url, true);
+      var code = url_parts.query.code;
+      if (url_parts.query.code) {
+        resolve(url_parts.query.code);
+      } else {
+        reject(url_parts.query.error);
+      }
+      res.end(LOG.AUTH_PAGE_SUCCESSFUL);
+      server.shutdown();
+    }).withShutdown();
+
+    server.listen(0, () => {
```

Comment:
Listen on port 0? I assume it's not in use. o.w. we'll get `ERR_SERVER_ALREADY_LISTEN`.

Reply:
0 just means pick an unused ephemeral port. For installed app flows, port is ignored for registered callback localhost URLs, specifically for this reason -- hardcoded ports can cause conflicts if there's already a service running on it.